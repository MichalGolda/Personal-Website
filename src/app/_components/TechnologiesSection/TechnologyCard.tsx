export type TechnologyCardProps = {
  title: string;
  content: () => React.ReactElement;
  icons?: React.ReactElement;
};

const TechnologyCard: React.FC<TechnologyCardProps> = ({
  title,
  content,
  icons,
}) => {
  return (
    <div className="w-full flex flex-col gap-y-4">
      <h4 className="text-lg md:text-xl lg:text-2xl text-secondary font-bold">
        {title}
      </h4>
      <p className="text-sm md:text-base text-body">{content()}</p>
      {icons && <div className="flex flex-row flex-wrap gap-4">{icons}</div>}
    </div>
  );
};

export default TechnologyCard;
