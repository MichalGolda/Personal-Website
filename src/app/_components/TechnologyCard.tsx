export type TechnologyCardProps = {
  title: string;
  content: string;
};

const TechnologyCard: React.FC<TechnologyCardProps> = ({ title, content }) => {
  return (
    <div className="w-full flex flex-col gap-y-4">
      <h4 className="text-2xl text-secondary font-bold">{title}</h4>
      <p className="text-body">{content}</p>
    </div>
  );
};

export default TechnologyCard;
