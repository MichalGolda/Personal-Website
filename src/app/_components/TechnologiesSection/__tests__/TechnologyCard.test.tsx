import "@testing-library/jest-dom";
import { render, screen } from "@testing-library/react";
import TechnologyCard from "../TechnologyCard";

describe("TechnologyCard component", () => {
  it("renders without errors", () => {
    const { container } = render(
      <TechnologyCard
        title="test title"
        content="test content"
        icons={<>test icons</>}
      />
    );

    const title = screen.getByText("test title");
    const content = screen.getByText("test content");
    const icons = container.firstChild?.lastChild;

    expect(title).toBeInTheDocument();
    expect(title.tagName).toBe("H4");

    expect(content).toBeInTheDocument();
    expect(content.tagName).toBe("P");

    expect(icons).toBeInTheDocument();
    expect(icons?.textContent).toBe("test icons");
  });

  it("renders unchanged", () => {
    const { container } = render(
      <TechnologyCard
        title="test title"
        content="test content"
        icons={<>test icons</>}
      />
    );

    expect(container).toMatchSnapshot();
  });
});
