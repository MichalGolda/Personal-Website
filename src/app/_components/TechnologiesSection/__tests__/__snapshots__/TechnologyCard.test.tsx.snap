// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TechnologyCard component renders unchanged 1`] = `
<div>
  <div
    class="w-full flex flex-col gap-y-4"
  >
    <h4
      class="text-lg md:text-xl lg:text-2xl text-secondary font-bold"
    >
      test title
    </h4>
    <p
      class="text-sm md:text-base text-body"
    >
      test content
    </p>
    <div
      class="flex flex-row flex-wrap gap-4"
    >
      test icons
    </div>
  </div>
</div>
`;
